### ISSUE TITLE
SQLite

### ISSUE THREAD

**Issue by weskerfoot** [CodeReaderMe/awesome-code-reading#17]
## Link to the codebase
[https://sqlite.org/src/dir?ci=tip](https://sqlite.org/src/dir?ci=tip)

## Which release/version did you read?
Latest release as of [https://sqlite.org/src/vinfo?name=427e2c9d95a92308](https://sqlite.org/src/vinfo?name=427e2c9d95a92308) (3.32.0)

## Why is it a good project to read?
* A full SQL database implemented in a relatively small number of lines of code
* High quality fundamental data structure implementations like BTrees and hash tables
* Good way to learn systems programming concepts and learn from a very high quality codebase in C

## Starting point
[sqlite.h.in](https://sqlite.org/src/artifact/facb0bbd7775449d)

## Reading order
* Start with the "source code tour" [here](https://sqlite.org/src/doc/trunk/README.md)
* sqlite.h.in and sqliteInt.h  to understand how it all fits together
* Basically follow the source code tour's suggestions
* BTree implementation

## Any parts should be skipped?
You don't need to understand the "amalgamation" code.

## Links and resources
* SQLite source code tour [https://sqlite.org/src/doc/trunk/README.md](https://sqlite.org/src/doc/trunk/README.md)
